src/mlpack/tests/lmnn_test.cpp

```diff
@@ -188,18 +185,18 @@ BOOST_AUTO_TEST_CASE(LMNNInitialEvaluateWithGradientTest)
   double objective = lmnnfn.EvaluateWithGradient(coordinates, gradient);
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(objective, 9.456, 1e-5);
+  REQUIRE(objective == Approx( 9.456).epsilon(1e-7));
```

Comment:
```suggestion
  REQUIRE(objective == Approx(9.456).epsilon(1e-7));
```

Remove the extra space here.